core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -208,6 +233,55 @@ public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, EncodingManager.A
         return edgeFlags;
     }
 
+    /**
+     * Calculate slopes from elevation data and set speed according to that. In-/declines between smallInclinePercent
+     * and maxInclinePercent will reduce speed to SLOW_SPEED. In-/declines above maxInclinePercent will result in zero
+     * speed.
+     * @param way
+     * @param edge
+     */
+    @Override
+    public void applyWayTags(ReaderWay way, EdgeIteratorState edge) {
+        PointList pl = edge.fetchWayGeometry(3);
+
+        double prevEle = pl.getElevation(0);
+        double fullDist2D = edge.getDistance();
+
+        if (Double.isInfinite(fullDist2D)) {
+            throw new IllegalStateException("Infinite distance should not happen due to #435. way ID=" + way.getId());
+        }
+
+        if (fullDist2D < 1) {
+            return;
+        }
+
+        double eleDelta = pl.getElevation(pl.size() - 1) - prevEle;
+        if (eleDelta > smallInclinePercent && eleDelta < maxInclinePercent) {
+            setFwdBwdSpeed(edge, SLOW_SPEED, MEAN_SPEED);
+        } else if (eleDelta > maxInclinePercent) {
+            setFwdBwdSpeed(edge, 0, 0);
+        } else if (eleDelta < -smallInclinePercent && eleDelta > -maxInclinePercent) {
+            setFwdBwdSpeed(edge, MEAN_SPEED, SLOW_SPEED);
+        } else if (eleDelta < -maxInclinePercent) {
+            setFwdBwdSpeed(edge, 0, 0);
+        }
+    }
+
+    /**
+     * Can set the given speed values to the given edge depending on the forward and backward accessibility of the edge.
+     * @param edge the edge to set speed for
+     * @param fwdSpeed speed value in forward direction
+     * @param bwdSpeed speed value in backward direction
+     */
+    private void setFwdBwdSpeed(EdgeIteratorState edge, int fwdSpeed, int bwdSpeed) {
+        if (accessEnc.getBool(false, edge.getFlags())) {
+            setSpeed(false, edge.getFlags(), fwdSpeed);
+        }
+        if (accessEnc.getBool(true, edge.getFlags())) {
+            setSpeed(true, edge.getFlags(), bwdSpeed);
+        }
+    }
```

Comment:
As the setSpeed method is deprecated we need a different approach here. Some more hints:

 * use edge.get(accessEnc) instead accessEnc.getBool
 * instead of setting speed to 0, set access to false

Reply:
The second case will fail if speed_two_directions=false.
Can you please add a test for this?

Reply:
Hmmh, but when we introduce it there it should also work for FootFlagEncoder. Let me know if you need help there.